lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/StorageLocationUtils.java

```
@@ -13,27 +14,36 @@
 public class StorageLocationUtils {
 
 	public static StorageLocationSetting convertDBOtoDTO(DBOStorageLocation dbo) {
-		StorageLocationSetting setting = dbo.getData();
-		setting.setStorageLocationId(dbo.getId());
-		setting.setDescription(dbo.getDescription());
-		setting.setUploadType(dbo.getUploadType());
-		setting.setEtag(dbo.getEtag());
-		setting.setCreatedBy(dbo.getCreatedBy());
-		setting.setCreatedOn(dbo.getCreatedOn());
-		return setting;
+		try {
+			StorageLocationSetting setting = EntityFactory.createEntityFromJSONString(dbo.getJson(),
+					StorageLocationSetting.class);
+			setting.setStorageLocationId(dbo.getId());
+			setting.setDescription(dbo.getDescription());
+			setting.setUploadType(UploadType.valueOf(dbo.getUploadType()));
+			setting.setEtag(dbo.getEtag());
+			setting.setCreatedBy(dbo.getCreatedBy());
+			setting.setCreatedOn(dbo.getCreatedOn());
+			return setting;
+		} catch (JSONObjectAdapterException e) {
```

We should probably throw a runtime exception in createEntityFromJsonString